hack/tilt/setup.sh

```diff
@@ -0,0 +1,239 @@
+#!/bin/bash
+
+# Copyright 2021 The Kubernetes Authors.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+TILT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+SCRIPT_DIR=$(dirname $TILT_DIR)
+BIN_DIR=$(dirname $SCRIPT_DIR)/temp/tilt/bin 
+
+KIND="kindest/node:v1.22.13@sha256:4904eda4d6e64b402169797805b8ec01f50133960ad6c19af45173a27eadf959"
+
+source ${SCRIPT_DIR}/utils.sh 
+source ${SCRIPT_DIR}/common.sh 
+
+
+
+function create_cluster {
+    ###########################################################################
+    # Description:                                                            #
+    # create cluster                                          #
+    #                                                                         #
+    # Arguments:                                                              #
+    #   arg1: ip_family                                                       #
+    ###########################################################################
+    [ $# -eq 1 ]
+    if_error_exit "set_host $# Wrong number of arguments to ${FUNCNAME[0]}"
+    local ip_family="${1}"
+    
+    ${BIN_DIR}/kind version
+
+    echo "****************************************************"
+    ${BIN_DIR}/kind delete cluster --name ${CLUSTER_NAME}
+
+    case $ip_family in
+        ipv4 )
+            CLUSTER_CIDR="${CLUSTER_CIDR_V4}"
+            SERVICE_CLUSTER_IP_RANGE="${SERVICE_CLUSTER_IP_RANGE_V4}"
+            ;;
+        ipv6 )
+            CLUSTER_CIDR="${CLUSTER_CIDR_V6}"
+            SERVICE_CLUSTER_IP_RANGE="${SERVICE_CLUSTER_IP_RANGE_V6}"
+            ;;
+        dual )
+            CLUSTER_CIDR="${CLUSTER_CIDR_V4},${CLUSTER_CIDR_V6}"
+            SERVICE_CLUSTER_IP_RANGE="${SERVICE_CLUSTER_IP_RANGE_V4},${SERVICE_CLUSTER_IP_RANGE_V6}"
+            ;;
+    esac
+    
+    # create the config file
+cat <<EOF >hack/tilt/kind.yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+    kubeProxyMode: "none"
+    apiServerAddress: "0.0.0.0"
+    disableDefaultCNI: true
+    ipFamily: "${ip_family}"
+    podSubnet: "${CLUSTER_CIDR}"
+    serviceSubnet: "${SERVICE_CLUSTER_IP_RANGE}"
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
+EOF
```

Comment:
I would recommend for a future patch to move this kind config to a external file and inside the this script execute a sed. Just trying to make it easier and not huge file.

Reply:
I'll submit a follow-up PR after this.